### PR TITLE
Feature/add macros

### PR DESCRIPTION
**Pull request by tbrezot**

### CODE REVIEW

**Comment by ThibaudDauce** [comment]
Is it possible to add `FindexSearch` / `FindexUpsert` / … inside the macro parameters to avoid having 3 similar macros?

**Comment by tbrezot** [comment]
Yes great idea, I did it in 5871e1f.

**Review by ThibaudDauce — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -98,92 +70,133 @@ cargo test --release --all-features
...
-Given `M` the average number of values indexed by keyword, the size of the
-compacted Chain Table in bytes is given by:
+Given `V(w_i)` the volume of the keyword `w_i` (i.e. the number of values
+indexed by this keyword), the size of the Chain Table in bytes after compacting
+is given by:
 
 ``` text
-N * ceil(M / TABLE_WIDTH) * (UID_LENGTH + ENCRYPTION_OVERHEAD + TABLE_WIDTH * BLOCK_LENGTH)
-= N * ceil(M / 5) * 220
+chain_table_size = sum_{w_i}(ceil(V(w_i) / CHAIN_TABLE_WIDTH)) * (UID_LENGTH + ENCRYPTION_OVERHEAD + 1 + CHAIN_TABLE_WIDTH * (1 + BLOCK_LENGTH))
```
Comment (Manuthor) [inline README.md:177]: use github latex equation?

File: README.md
Comment (Manuthor) [inline README.md:1]: could you create a separate markdown file to describe Findex callbacks and link in the README?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,19 @@
 
 All notable changes to this project will be documented in this file.
 
+## Unreleased
+
+### Changed
+
+- deletions in upsert
+- format of the Chain Table
```
[the remark this answers is not in the document]
Comment (tbrezot) [reply CHANGELOG.md]: done in 0c9dd8b

File: README.md
[the remark this answers is not in the document]
Comment (tbrezot) [reply README.md:1]: fixed in 7a86997

File: README.md [hunk 3]
```diff
@@ -98,92 +70,133 @@ cargo test --release --all-features
...
+  operation.
+- the UID stored in the Entry Table corresponds to the last Chain Table UID
+  that stores values indexed by `w_i`.
+- the blocks stored in the Chain Table are used to pad the Chain Table lines to
+  an equal size. Each block is 16-bytes long (which is enough to store an
+  UUID). An indexed value is stored in the Chain Table by chunks of 16 bytes.
+  The last chunk may not be full and is padded with 0s.
+- the number of blocks per Chain Table value (`B`) used is 5.
+
+The Chain Table values are serialized as follows (sizes are given in bytes):
```
[the remark this answers is not in the document]
Comment (tbrezot) [reply README.md:134]: This is the serialization before encryption.

File: README.md (hunk 1 above)
Comment (tbrezot) [reply README.md]: done

File: README.md [hunk 4]
```diff
@@ -57,37 +57,9 @@ To build Findex without the interfaces, run:
...
-```bash
-./python/scripts/test.sh
-```
-
-And finally, to build everything and test it, run:
-
-```bash
-cargo build --release --all-features
 cargo test --release --all-features
 ```
```
[the remark this answers is not in the document]
Comment (tbrezot) [reply README.md:65]: Technically, benchmarks are a subsection of this section... they just are at the end of it.

**Review by Manuthor — APPROVED** [review]

**Review by Manuthor — APPROVED** [review]

**Pull request merged**